client/consensus/slots/src/lib.rs

```diff
@@ -583,9 +600,86 @@ pub fn slot_lenience_linear(parent_slot: u64, slot_info: &SlotInfo) -> Option<Du
 	}
 }
 
+/// Trait for providing the strategy for when to backoff block authoring.
+pub trait BackoffAuthoringBlocksStrategy<N>
+where
+	N: BaseArithmetic
+{
```

Comment:
```suggestion
pub trait BackoffAuthoringBlocksStrategy<N> {
```

The bound is not required.